### PR TITLE
Update openssl3 in pypi

### PR DESCRIPTION
**Pull request by phoebusm**
#### Reference Issues/PRs

#1491 
#### What does this implement or fix?
Update openssl to v3
#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

File: build_tooling/build_many_linux_image.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ RUN rpmkeys --import 'https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x3F
             s/#?exclude.*/exclude=.edu/' /etc/yum/pluginconf.d/fastestmirror.conf
 ADD sccache /usr/local/bin/
 RUN yum update -y && \
-    yum install -y zip jq less devtoolset-11-gdb \
+    yum install -y zip jq less devtoolset-11-gdb perl-IPC-Cmd \
```
Comment (poodlewars) [inline build_tooling/build_many_linux_image.sh:47]: Why do we need `perl-IPC-Cmd` ?

Comment (phoebusm) [reply build_tooling/build_many_linux_image.sh:47]: It is required by v3:
https://github.com/man-group/ArcticDB/actions/runs/8649691810/job/23716646005
```
Can't locate IPC/Cmd.pm in @INC (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at -e line 1.
  BEGIN failed--compilation aborted at -e line 1.
  CMake Error at buildtrees/versioning_/versions/openssl/53fb72a19981a3439cd569f58b722ca6cd26e23d/unix/portfile.cmake:27 (message):
    
  
    Perl cannot find IPC::Cmd.  Please install it through your system package
    manager.
```

**Comment by poodlewars** [comment]
Change looks fine. What test coverage do we have for our SSL interaction? Do the persistent storage tests use HTTPS? Worth running them if so, or doing a quick manual test if not.

**Comment by phoebusm** [comment]
> Change looks fine. What test coverage do we have for our SSL interaction? 

https://github.com/man-group/ArcticDB/pull/1338
S3 auto ca path PR has already been merged. It contains the https related tests.

> Do the persistent storage tests use HTTPS? Worth running them if so, or doing a quick manual test if not.
Good point. Let me turn it on

**Comment by phoebusm** [comment]
> Do the persistent storage tests use HTTPS? Worth running them if so, or doing a quick manual test if not.

@poodlewars It is on now.

**Comment by poodlewars** [comment]
> > Do the persistent storage tests use HTTPS? Worth running them if so, or doing a quick manual test if not.
> 
> @poodlewars It is on now.

@phoebusm can you share a link showing it passing please?

**Comment by phoebusm** [comment]
@poodlewars 
```
root@XPS:/tmp# cat /etc/*-release | grep PRETTY_NAME
PRETTY_NAME="Ubuntu 22.04.4 LTS"
root@XPS:/tmp# python -m venv venv
root@XPS:/tmp# source venv/bin/activate
(venv) root@XPS:/tmp# pip install arcticdb-4.1.0.dev0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl 
(venv) root@XPS:/tmp# python
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arcticdb as adb
>>> arctic = adb.Arctic("s3s://s3-eu-north-1.amazonaws.com:XXXXXXXXXXXX?aws_auth=true&ssl=True")
>>> arctic.list_libraries()
[]
>>> quit()
(venv) root@XPS:/tmp# ls -al /etc/pki/ca-trust/extracted/pem/
ls: cannot access '/etc/pki/ca-trust/extracted/pem/': No such file or directory
(venv) root@XPS:/tmp# ls -al /etc/ssl/certs/ca-certificates.crt
-rw-r--r-- 1 root root 208567 Nov 27 14:45 /etc/ssl/certs/ca-certificates.crt
```

**Review by poodlewars — APPROVED** [review]

**Pull request merged**